### ISSUE TITLE
get_confidence函数中随着迭代过程，confidence / num_tokens不应该是越低越好吗，如果logits分布比较均匀，confidence / num_tokens应该会很高吧

### ISSUE THREAD

**Issue by ljzx20** [ltpo2025/LTPO#2]
def get_confidence(
    model,
    inputs,
    thought_idx,
    thought_hidden_states,
    k=10,
):
    inputs['inputs_embeds'][0, thought_idx[0]:thought_idx[1]] = thought_hidden_states
    logits = model(**inputs, return_dict=True)['logits'][0]
    probs = torch.softmax(logits, dim=-1)
    confidence = 0.0
    for idx in range(thought_idx[0], thought_idx[1] + 1):
        topk = torch.topk(probs[idx], k=k, largest=True)[0]
        confidence -= torch.sum(torch.log(topk + 1e-10)) / k
    num_tokens = thought_idx[1] - thought_idx[0] + 1
    return confidence / num_tokens

**Comment by ltpo2025** [comment]
Hi @ljzx20 谢谢你的问题！为了直观地解释，我们简化一下，假设只有5个tokens，我们分别计算不同分布下的confidence值：
```
均匀分布:
(0.20, 0.20, 0.20, 0.20, 0.20) -> -(log0.20 + log0.20 + log0.20 + log0.20 + log0.20) / 5 ~= 1.609

较弱peaked:
(0.40, 0.20, 0.15, 0.15, 0.10) -> -(log0.40 + log0.20 + log0.15 + log0.15 + log0.10) / 5 ~= 1.725

较强peaked:
(0.70, 0.12, 0.08, 0.06, 0.04) -> -(log0.70 + log0.12 + log0.08 + log0.06 + log0.04) / 5 ~= 2.207

极强peaked:
(0.90, 0.04, 0.03, 0.02, 0.01) -> -(log0.90 + log0.04 + log0.03 + log0.02 + log0.01) / 5 ~= 3.070
```

另外，值得一提的是 Deep Think with Confidence ([Fu et al., 2026](https://openreview.net/forum?id=8LqHs0KIM7)) 里详细地剖析了confidence的作用。
